recipes-extended/xen/files/allow-stubdoms-cacheattr-control.patch

```diff
@@ -77,17 +77,17 @@ PATCHES
 ################################################################################
 --- a/xen/include/xsm/dummy.h
 +++ b/xen/include/xsm/dummy.h
-@@ -127,6 +127,7 @@ static XSM_INLINE int xsm_domctl(XSM_DEF
+@@ -141,6 +141,7 @@ static XSM_INLINE int xsm_domctl(XSM_DEF
      case XEN_DOMCTL_memory_mapping:
      case XEN_DOMCTL_bind_pt_irq:
      case XEN_DOMCTL_unbind_pt_irq:
-+    case XEN_DOMCTL_pin_mem_cacheattr:
++    case XEN_DMOP_pin_memory_cacheattr:
```

Comment:
I think this should be dropped since it's covered by dm_op.  There is a separate xsm_dm_op function in `xen/include/xsm/dummy.h` for the dm_ops.  It is already 
```
    XSM_ASSERT_ACTION(XSM_DM_PRIV);
    return xsm_default_action(action, current->domain, d);
```
so pin_memory_cacheattr is already covered the same way.

Also, XEN_DOMCTL_pin_mem_cacheattr (41) != XEN_DMOP_pin_memory_cacheattr (18), so this is probably allowing XEN_DOMCTL_setdebugging (18)

Reply:
Good catch, will drop.
Do you understand the second part of that patch by any chance? Should that be kept?

Reply:
I can only go from the description.  @dozylynx re-ordered the code - I guess to get to hvm_get_mem_pinned_cacheattr before the `if ( !need_iommu(d) && !cache_flush_permitted(d) )` check.  That way memory can be set to write combine - otherwise it would be write back.  So I guess it's still needed.